stdlib/ballerina-builtin/src/main/ballerina/caching/caching.bal

```diff
@@ -104,23 +104,36 @@ public type Cache object {
     an error will be thrown. So use the hasKey function to check for the existance of a particular key."}
     @Param {value:"key: key which is used to retrieve the cached value"}
     @Return {value:"The cached value associated with the given key"}
-    public function get (string key) returns any? {
-        // If the key is not found, an error will be thrown from here.
+    public function get(string key) returns any? {
+        // Check whether the required cache is available.
+        if (!self.hasKey(key)){
+            return ();
+        }
+        // Get the required cache from the map.
         any value = self.entries[key];
-        // So the value will be always not null here.
         var entry = <CacheEntry>value;
+        // Check whether the cache is already expired. Since the cache cleaning task runs in predefined intervals,
+        // somethimes the cache might not be expired. So this gurentees that the expired caches will not be returened.
```

Comment:
s/somthimes/somtimes

Reply:
This comment is not correct. It is not that the cache is expired, the cache entry is expired.